### PR TITLE
oneclick: tithe farm improvements

### PR DESCRIPTION
**Pull request by ocekyled**

### CODE REVIEW

File: OneClick/src/main/java/net/runelite/client/plugins/externals/oneclick/OneClickPlugin.java [hunk 1]
```diff
@@ -455,6 +460,7 @@ private void onMenuEntryAdded(MenuEntryAdded event)
 				}
 				break;
 			case SEED_SET:
+				Widget titheWidget = client.getWidget(WidgetInfo.TITHE_FARM);
```
Comment (Ganom) [inline OneClick/src/main/java/net/runelite/client/plugins/externals/oneclick/OneClickPlugin.java]: That is an insane amount of objects generated by not having this locked behind a guard clause, you should ensure the opcode is what you want, and the target, then and only then check the widget.

Comment (ocekyled) [reply OneClick/src/main/java/net/runelite/client/plugins/externals/oneclick/OneClickPlugin.java]: Sorry, I don't understand at all. I looked through other plugins that use similar checks to see if a widget is hidden and copied that logic. I''m very new to working with java, any chance you could elaborate some more on how this should be handled?

Comment (Ganom) [reply OneClick/src/main/java/net/runelite/client/plugins/externals/oneclick/OneClickPlugin.java]:
```java
else if (event.getOpcode() == MenuOpcode.WALK.getId())
{
	if (titheWidget == null || titheWidget.isHidden())
	{
		return;
	}
	MenuEntry menuEntry = client.getLeftClickMenuEntry();
	menuEntry.setOpcode(MenuOpcode.WALK.getId() + MENU_ACTION_DEPRIORITIZE_OFFSET);
}
```

Comment (ocekyled) [reply OneClick/src/main/java/net/runelite/client/plugins/externals/oneclick/OneClickPlugin.java]: Ok that's basically what I thought you meant, but it seemed too simple to be the correct answer.
Thank you for your help

**Pull request merged**